examples/tutorials/flow-deploy/kubernetes/README.md

```diff
@@ -0,0 +1,153 @@
+# Deploy flow using Kubernetes
+This example demos how to deploy flow as a Kubernetes app. 
+We will use [web-classification](../../../flows/standard/web-classification/README.md) as example in this tutorial.
+
+Please ensure that you have installed all the required dependencies. You can refer to the "Prerequisites" section in the README of the [web-classification](../../../flows/standard/web-classification/README.md#Prerequisites) for a comprehensive list of prerequisites and installation instructions.
+
+## Build a flow as docker format
+
+Note that all dependent connections must be created before building as docker.
+```bash
+# create connection if not created before
+pf connection create --file ../../../connections/azure_openai.yml --set api_key=<your_api_key> api_base=<your_api_base> --name open_ai_connection
+```
+
+Use the command below to build a flow as docker format app:
+
+```bash
+pf flow build --source ../../../flows/standard/web-classification --output build --format docker
+```
+
+## Deploy with Kubernetes
+### Build Docker image
+
+Like other Dockerfile, you need to build the image first. You can tag the image with any name you want. In this example, we use `web-classification-serve`.
+
+Then run the command below:
+
+```shell
+cd build
+docker build . -t web-classification-serve
+```
+
+### Create Kubernetes deployment yaml.
+The Kubernetes deployment yaml file acts as a guide for managing your docker container in a Kubernetes pod. It clearly specifies important information like the container image, port configurations, environment variables, and various settings. Below, you'll find a simple deployment template that you can easily customize to meet your needs.
+
+You need encode the secret using base64 firstly and input the encoded value as 'open-ai-connection-api-key' in the deployment configuration, for example:
+```shell
+encoded_secret=$(echo -n <your_api_key> | base64)
+```
+
+```yaml
+---
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: web-classification
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: open-ai-connection-api-secret
+  namespace: web-classification
+type: Opaque
+data:
+  open-ai-connection-api-key: <encoded_secret>
```

Comment:
this is the only thing which need to be changed manually by user right?

Reply:
Not sure if we should run this secret creation as another command

Reply:
there're some values need  to be changed manually by user :
1. encoded secret
2. pod_name
3. If user choose option2 to curl endpoint, they still need run `minikube service web-classification-service --url -n web-classification` to get the assigned port. 

As far as I know, even though I expose port 30123 via NodePort in minikube, minikube will still not expose it because it will use its own external port to listen to this service. Minikube tunnels the service to expose to the outer world.

Reply:
I see, but that's the limitation of minikube?  not for a full fledged kubernettes right?